subsys/usb/host/class/usbh_hub_mgr.c

```diff
@@ -0,0 +1,1166 @@
+	hub_mgr_data->int_active = true;
+	return 0;
+}
+
+/**
+ * @brief Start hub interrupt monitoring
+ */
+static int usbh_hub_mgr_start_interrupt(struct usbh_hub_mgr_data *hub_mgr_data)
+{
+	struct uhc_transfer *xfer;
+	struct net_buf *buf;
+	int ret;
+
+	if (!hub_mgr_data || hub_mgr_data->being_removed || hub_mgr_data->int_active) {
+		return -EINVAL;
+	}
+
+	/* Start interrupt only when operational */
+	if (hub_mgr_data->state != HUB_STATE_OPERATIONAL) {
+		return -ENOENT;
+	}
+
+	/* Check interrupt endpoint */
+	if (!hub_mgr_data->int_ep) {
+		LOG_ERR("No interrupt endpoint available");
+		return -ENODEV;
+	}
+
+	/* Allocate interrupt transfer */
+	xfer = usbh_xfer_alloc(hub_mgr_data->hub_udev, hub_mgr_data->int_ep->bEndpointAddress,
+			       usbh_hub_mgr_interrupt_in_cb, (void *)hub_mgr_data);
+	if (!xfer) {
+		LOG_ERR("Failed to allocate interrupt transfer");
+		return -ENOMEM;
+	}
+
+	/* Allocate receive buffer */
+	buf = usbh_xfer_buf_alloc(hub_mgr_data->hub_udev, 
+				  sys_le16_to_cpu(hub_mgr_data->int_ep->wMaxPacketSize));
+	if (!buf) {
+		LOG_ERR("Failed to allocate interrupt buffer");
+		usbh_xfer_free(hub_mgr_data->hub_udev, xfer);
+		return -ENOMEM;
+	}
+
+	xfer->buf = buf;
+
+	ret = usbh_xfer_enqueue(hub_mgr_data->hub_udev, xfer);
+	if (ret != 0) {
+		LOG_ERR("Failed to enqueue interrupt transfer: %d", ret);
+		usbh_xfer_buf_free(hub_mgr_data->hub_udev, buf);
+		usbh_xfer_free(hub_mgr_data->hub_udev, xfer);
+		return ret;
+	}
+
+	hub_mgr_data->interrupt_transfer = xfer;
+	hub_mgr_data->int_active = true;
+	hub_mgr_data->prime_status = HUB_PRIME_INTERRUPT;
+	
+	LOG_DBG("Hub level %d interrupt monitoring started", hub_mgr_data->hub_level);
+	return 0;
+}
+
+/**
+ * @brief Find hub manager data by USB device
+ */
+static struct usbh_hub_mgr_data *find_hub_mgr_by_udev(struct usb_device *udev)
+{
+	struct usbh_hub_mgr_data *hub_mgr_data;
+	
+	k_mutex_lock(&hub_mgr.lock, K_FOREVER);
+	
+	SYS_SLIST_FOR_EACH_CONTAINER(&hub_mgr.hub_list, hub_mgr_data, node) {
+		if (hub_mgr_data->hub_udev == udev) {
+			k_mutex_unlock(&hub_mgr.lock);
+			return hub_mgr_data;
+		}
+	}
+	
+	k_mutex_unlock(&hub_mgr.lock);
+
+	return NULL;
+}
+
+/**
+ * @brief Process hub interrupt data
+ */
+static void usbh_hub_mgr_process_data(struct usbh_hub_mgr_data *hub_mgr_data)
+{
+	uint8_t port_index;
+
+	k_mutex_lock(&hub_mgr_data->lock, K_FOREVER);
+
+	if (hub_mgr_data->being_removed) {
+		k_mutex_unlock(&hub_mgr_data->lock);
+		return;
+	}
+
+	for (port_index = 0; port_index <= hub_mgr_data->num_ports; ++port_index) {
+		if (0 != ((0x01U << (port_index & 0x07U)) & 
+			  (hub_mgr_data->int_buffer[port_index >> 3U]))) {
+
+			if (port_index == 0) {
+				/* Hub status change */
+				LOG_INF("Hub level %d status changed, processing", hub_mgr_data->hub_level);
+				int ret = usbh_hub_get_hub_status(&hub_mgr_data->hub_instance, 
+								&hub_mgr_data->hub_instance.hub_status.wHubStatus,
+								&hub_mgr_data->hub_instance.hub_status.wHubChange);
+				if (ret == 0) {
```

Comment:
How about using `if (ret != 0) { LOG_ERR(); continue; }` to avoid adding too many nested levels of `if { ... }`?